### PR TITLE
Changes for the tidyverse port

### PR DESCRIPTION
**Pull request by lionel-**
Includes #48 

Various changes needed for porting tibble, dplyr and tidyr to tidyeval.

### CODE REVIEW

**Comment by lionel-** [comment]
Now uses the "quosure" and "overscope" vocabulary.

File: R/env.R [hunk 1]
```diff
@@ -552,6 +595,26 @@ env_clone <- function(x, parent = env_parent(x)) {
   list2env(as.list(x, all.names = TRUE), parent = parent)
 }
 
+#' Does environment inherits from another environment?
```
Comment (lionel-) [inline R/env.R]: This is not consistent with how we have written titles so far, e.g. all those "Is object an xxx" titles for the type predicates. Should we adopt this style from now on?

File: R/quosure.R [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+#'
+#'
+#' # Formulas and definitions are not quosures:
+#' is_quosure(a := b)
+#' is_quosure(a ~ b)
+#'
+#' # But they are quosureish objects:
+#' is_quosureish(a := b)
+#' is_quosureish(a ~ b)
+#' @md
```
Comment (lionel-) [inline R/quosure.R]: Yes I was planning to port the whole codebase to markdown once all branches are merged.

File: R/env.R (hunk 1 above)
Comment (hadley) [inline R/env.R]: I was just correcting your English, not making a general statement about grammatical forms :wink:

Comment (lionel-) [inline R/env.R]: But I got this grammatical form from your early purrr documentation ;)

it feels a bit like contracted newspaper titles so it made sense to me.

Comment (hadley) [inline R/env.R]: Well you can keep it, but definitely needs "inherits" -> "inherit"

Comment (lionel-) [inline R/env.R]: oh yes sorry

**Pull request merged**